algebird-core/src/main/scala/com/twitter/algebird/statistics/GaussianDistributionMonoid.scala

```diff
@@ -0,0 +1,28 @@
+package com.twitter.algebird.statistics
+
+import com.twitter.algebird.Monoid
+
+/**
+ * @mean - Mean
+ * @sigma - Standard deviation
+ * aka Normal distribution
+ */
+case class GaussianDistribution(mean: Double, sigma: Double)
+
+object GaussianDistribution {
+  implicit def monoid: Monoid[GaussianDistribution] = new GaussianDistributionMonoid
+}
+
+/**
+ * This monoid stems from the fact that if X and Y are independent random variables
+ * that are normally distributed, then their sum is also
+ * normally distributed, with its new mean equal to the sum of two means
+ * and variance equal to the sum of two variances.
+ */
+class GaussianDistributionMonoid extends Monoid[GaussianDistribution] {
+  override def zero = new GaussianDistribution(0, 0)
+
+  override def plus(left: GaussianDistribution, right: GaussianDistribution): GaussianDistribution = {
+    new GaussianDistribution(left.mean + right.mean, left.sigma + right.sigma)
+  }
```

Comment:
please add an optimized sumOption:

```scala
def sumOption(its: TraversableOnce[GaussionDistribution]): Option[GaussionDistribution] =
  if (its.isEmpty) None
  else {
    var mean = 0.0
    var sigma = 0.0
    val it = its.toIterator
    while (it.hasNext) {
      val g = it.next
      mean += g.mean
      sigma += g.sigma
    }
    GuassianDistribution(mean, sigma)
  }
```
The point is that summing on a large list can avoid a lot of allocation using .sum or .sumOption, which this method will enable. It uses `vars` but only locally, and still gives the user a nice, immutable functional API.